### ISSUE TITLE
Warnings due to unmet peer dependency typescript

### ISSUE THREAD

**Issue by vieira** [typescript-eslint/typescript-eslint#1287]
Hello,

I use `eslint-plugin-jest` which has a dependency to `@typescript-eslint/experimental-utils`. Since #1120 yarn started emitting warnings:

```
warning "eslint-plugin-jest > @typescript-eslint/experimental-utils > @typescript-eslint/typescript-estree > tsutils@3.17.1" has unmet peer dependency "typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta"
```

As I understand from jest-community/eslint-plugin-jest#467 other projects expect to use  `@typescript-eslint/typescript-estree` without requiring TS by consumers.

In my case, I use jest and `eslint-plugin-jest` in a project without typescript.

**Comment by bradzacher** [comment]
You can ignore this warning.
There's nothing we can really do about this right now.

Could potentially convince tsutils to add optional peer dep meta config, but I don't know if they'd want to do that.
It would take a breaking change to restructure the dependency tree of the packages.
